scripts/deployCollector.ts

```diff
@@ -0,0 +1,159 @@
+import * as fs from 'fs';
+
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { Collector } from 'typechain-types';
+
+export const DEFAULT_CONFIG_FILE_NAME = 'deploy-collector.input.json';
+export const DEFAULT_OUTPUT_FILE_NAME = 'revenue-sharing-addresses.json';
+
+export interface Partner {
+  beneficiary: string;
+  share: number;
+}
+
+interface CollectorConfig {
+  collectorOwner: string;
+  partners: Partner[];
+  tokenAddress: string;
+  remainderAddress: string;
+}
+
+export type DeployCollectorArg = {
+  collectorConfigFileName?: string;
+  outputFileName?: string;
+};
+
+type ChainId = string;
+type OutputCollectorConfig = CollectorConfig | { collectorContract: string };
+export type OutputConfig = Record<ChainId, OutputCollectorConfig>;
+
+export const deployCollector = async (
+  taskArgs: DeployCollectorArg,
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { ethers, network } = hre;
+
+  if (!network) {
+    throw new Error('Unknown Network');
+  }
+
+  let collectorConfigFileName = taskArgs.collectorConfigFileName;
+
+  if (!collectorConfigFileName) {
+    console.warn(
+      "Missing '--collectorConfig' parameter, 'deploy-collector.input.json' will be used"
+    );
+    collectorConfigFileName = DEFAULT_CONFIG_FILE_NAME;
+  }
+
+  if (!fs.existsSync(collectorConfigFileName)) {
+    throw new Error('Could not find Collector configuration file');
+  }
+
+  const inputConfig = JSON.parse(
+    fs.readFileSync(collectorConfigFileName, { encoding: 'utf-8' })
+  ) as CollectorConfig;
+
+  const { collectorOwner, partners, tokenAddress, remainderAddress } =
+    inputConfig;
+
+  const collectorFactory = await ethers.getContractFactory('Collector');
+  const collector = await collectorFactory.deploy(
+    collectorOwner,
+    tokenAddress,
+    partners,
+    remainderAddress
+  );
```

Comment:
Not important, just an FYI: you should be able to use `Collector_factory` class from type-chain directly instead of specifying it by name string.

```suggestion
  const collectorFactory = new Collector__factory();
  const collector = await collectorFactory.deploy(
    collectorOwner,
    tokenAddress,
    partners,
    remainderAddress
  );
```

Reply:
I think that the usage of `await ethers.getContractFactory('Collector');` is easier than `new Collector__factory(contractInterface: ContractInterface, bytecode: BytesLike | { object: string }, signer?: Signer)` and I don't see any benefit in using it. Am i missing anything?